msl/equipment/resources/vaisala/ptu300.py

```diff
@@ -0,0 +1,194 @@
+"""
+Vaisala Barometer which reads temperature, relative humidity, and pressure, e.g. of type PTU300
+"""
+from __future__ import annotations
+
+import re
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from msl.equipment import EquipmentRecord
+from msl.equipment.resources import register
+from msl.equipment.exceptions import VaisalaError
+from msl.equipment.connection_serial import ConnectionSerial
+
+from msl.equipment.constants import (
+    Parity,
+    DataBits,
+)
+
+
+@register(manufacturer=r'Vaisala', model=r'PTU30*', flags=re.IGNORECASE)
```

Comment:
Tip: In _regex_ the `*` character means "the preceding character zero or more times". It is not a generic wildcard character like commonly used in file-system paths. So in your case, the model must match the characters `PTU3` and `0` zero or more times. Perhaps you want to use `\d` instead of `*` or delete the `*` or if only models that contain the characters PTU300, PTU301, PTU303, PTU307 are supported then use `PTU30[0137]`.

`\d`  matches a single digit (equivalent to [0-9])

I typically have not been concerned with using `^` (start with) and `$` (end with) anchors, but we can start being more restrictive. So,  `^PTU30[0137]$` would match `PTU301` but not `PTU301A`.

Reply:
Hmmm ok, probably [0137T] would be the best option for the PTU300 class. I think the T option would be supported but I have no way of testing this without that specific device.